From: pl@openoffice.org
To: cvs@gsl.openoffice.org
Subject: [gsl-cvs] CVS update [cws_src680_vcl56]: /gsl/vcl/unx/gtk/window/

Tag: cws_src680_vcl56
User: pl      
Date: 06/03/16 04:28:11

Modified:
 /gsl/vcl/unx/gtk/window/
  gtkframe.cxx

Log:
 #i63086# workaround another metacity bug

File Changes:

Directory: /gsl/vcl/unx/gtk/window/
===================================

File [changed]: gtkframe.cxx
Url: http://gsl.openoffice.org/source/browse/gsl/vcl/unx/gtk/window/gtkframe.cxx?r1=1.42&r2=1.42.88.1
Delta lines:  +17 -3
--------------------
--- vcl/unx/gtk/window/gtkframe.cxx	25 Jan 2006 11:40:22 -0000	1.42
+++ vcl/unx/gtk/window/gtkframe.cxx	16 Mar 2006 12:28:08 -0000	1.42.88.1
@@ -1005,6 +1005,20 @@
             if( m_pParent && m_pParent->m_nWorkArea != m_nWorkArea )
                 getDisplay()->getWMAdaptor()->switchToWorkArea( m_pParent->m_nWorkArea );
             
+            if( isFloatGrabWindow() &&
+                m_pParent &&
+                m_nFloats == 0 &&
+                ! getDisplay()->GetCaptureFrame() )
+            {
+                /* #i63086#
+                 * outsmart Metacity's "focus:mouse" mode
+                 * which insists on taking the focus from the document
+                 * to the new float. Grab focus to parent frame BEFORE
+                 * showing the float (cannot grab it to the float
+                 * before show).
+                 */
+                 m_pParent->grabPointer( TRUE, TRUE );
+            }
             gtk_widget_show( GTK_WIDGET(m_pWindow) );
             if( isFloatGrabWindow() )
             {




